parsers/SG.py

```diff
@@ -1,11 +1,12 @@
 #!/usr/bin/env python3
 
 from collections import defaultdict
+import logging
```

Comment:
Minor comment, since this only used once maybe it should be `from logging import getLogger`.

Reply:
Hm, this is a style thing ([there is no perf difference](https://softwareengineering.stackexchange.com/questions/187403/import-module-vs-from-module-import-function/187471)). My very personal reason for liking `logging.getLogger()` is because I don't have to scroll up to see if `getLogger` was imported and from which module, or maybe a function somewhere within this file. To give an extreme example, if we were to do `from re import search`, it's impossible for someone diving straight into the code to guess what `search` is without reading all imports.

I don't think we have particular consistency in the codebase on this right now. At the same time I myself did the `from collections import defaultdict` right above :D

Reply:
Current codebase: `import logging` 9 times including SG (and in all cases it is only used for `getLogger`), `from logging import getLogger` 7 times. So if I change it, it'll be a tie. Do we have a style guide? :D

Reply:
Oh I thought there was a difference, never mind then it doesn't matter. If there was an EM style guide it would simply read "Chaos is good". :smiling_imp:

Reply:
"Carbon is bad, chaos is good"